types/rails__actioncable/index.d.ts

```diff
@@ -0,0 +1,159 @@
+// Type definitions for @rails/actioncable 6.1
+// Project: https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable
+// Definitions by: Martin Badin <https://github.com/martin-badin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace ActionCable;
+
+export enum MessageTypes {
+    confirmation = 'confirm_subscription',
+    disconnect = 'disconnect',
+    ping = 'ping',
+    rejection = 'reject_subscription',
+    welcome = 'welcome',
+}
+
+export enum DisconnectReasons {
+    invalid_request = 'invalid_request',
+    server_restart = 'server_restart',
+    unauthorized = 'unauthorized',
+}
+
+/**
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/internal.js
+ */
+export const INTERNAL: {
+    default_mount_path: '/cable';
+    disconnect_reasons: typeof DisconnectReasons;
+    message_types: typeof MessageTypes;
+    protocols: ['actioncable-v1-json', 'actioncable-unsupported'];
+};
+
+/**
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/connection.js
+ */
+export class Connection {
+    constructor(consumer: Consumer);
+
+    send(data: object): boolean;
+
+    open(): boolean;
+
+    close(options?: { allowReconnect: boolean }): any;
+
+    reopen(): void;
+
+    getProtocol(): void | string;
+
+    isOpen(): boolean;
+
+    isActive(): boolean;
+
+    static readonly reopenDelay: number;
+
+    static readonly events: {
+        message(event: {
+            data: {
+                identifier: string;
+                message: string;
+                reason: DisconnectReasons;
+                reconnect: boolean;
+                type: MessageTypes;
+            };
+        }): any;
+
+        open(): void;
+
+        close(): void;
+
+        error(): void;
+    };
+}
+
+/**
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/connection_monitor.js
+ */
+export class ConnectionMonitor {
+    constructor(connection: Connection);
+
+    start(): void;
+
+    stop(): void;
+
+    isRunning(): boolean;
+
+    recordPing(): void;
+
+    recordConnect(): void;
+
+    recordDisconnect(): void;
+
+    static readonly staleThreshold: number;
+
+    static readonly reconnectionBackoffRate: number;
+}
+
+/**
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/consumer.js
+ */
+export class Consumer {
+    constructor(url: string);
+
+    url: string;
+
+    send(data: object): boolean;
+
+    connect(): boolean;
+
+    disconnect(): any;
+
+    ensureActiveConnection(): void | boolean;
+}
+
+/**
+ * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/subscription.js
+ */
+export class Subscription {
+    constructor(consumer: Consumer, params?: object, mixin?: any);
+
+    perform(action: string, data?: object): boolean;
+
+    send(data: object): boolean;
+
+    unsubscribe(): this;
```

Comment:
interesting notation

Reply:
The function returns the same object as you want to unsubscribe

Reply:
Yeah, I got it, but didn't see it in action before :D